jiant/tasks/lm.py

```diff
@@ -181,14 +181,17 @@ def __init__(self, path, *args, **kw):
         }
 
 
-@register_task("mlm", rel_path="WikiText103/")
+@register_task("wikipedia_corpus_mlm", rel_path="wikipedia_corpus_small/")
 class MaskedLanguageModelingTask(Task):
     """
     Masked language modeling task on Wikipedia dataset
     Attributes:
         max_seq_len: (int) maximum sequence length
         min_seq_len: (int) minimum sequence length
         files_by_split: (dict) files for three data split (train, val, test)
+    We are currently using an unpreprocessed version of the Wikipedia corpus
+    that consists of 5% of the data. Please reach out to jiant admin if you 
+    would like access to this dataset.
```

Comment:
Is the task registered as `wikipedia_corpus_mlm` (using dataset under relative path `/wikipedia_corpus_small`) an experiment with some research significance that users would want to reproduce (or is it perhaps a toy dataset being used to demo `MaskedLanguageModelingTask`)?

Reply:
Oops, looks like I commented on some outdated code. I'll look at the updated code and refile my comment if necessary.

Reply:
It's not toy data. The full Wikipedia is very large and jiant currently does not have ability to load the entire thing. So, I tried to extract a subset with the same size as Wikitext 103, which is around 5% of full Wikipedia.

Reply:
Thanks for clarifying, @phu-pmh. Maybe relevant: there was a [recent discussion](https://github.com/nyu-mll/jiant/pull/1030#discussion_r398754495) and [change](https://github.com/nyu-mll/jiant/pull/1030/commits/8ca1eba4828362beda6f0436c2585f768ef2c28c) to another LM task to reduce its memory footprint. It may be relatively easy to make a similar change here to allow you to use the full dataset (if memory, not time, is the concern).

But if you want to introduce a modified dataset with this task, please also submit the code you used to construct the dataset. There's an example of a data preprocessing script under [`jiant/scripts`](https://github.com/nyu-mll/jiant/blob/master/scripts/winograd/preprocess_winograd.py). But all your script would need to do is document/link the data it takes as input, perform your preprocessing, and save the output as it's expected by your task.

Reply:
The preprocessing code we used was a slight modification of what is here (@phu-pmh 
can speak more to this), so it might not be the best idea to copy paste it into scripts. https://github.com/NVIDIA/DeepLearningExamples/tree/master/PyTorch/LanguageModeling/BERT/data . Perhaps we could put instructions (maybe in the documentation in the Task class?) about how to generate it instead.

Reply:
For English Wikipedia, I think the nvidia code is ready to use. I only added modifications so that we can do cross-lingual experiments.